tracing/tracing.hh

```diff
@@ -665,7 +667,7 @@ private:
 
 void one_session_records::set_pending_for_write() {
     _is_pending_for_write = true;
-    budget_ptr = tracing::get_local_tracing_instance().get_pending_records_ptr();
+    budget_ptr = _local_tracing_ptr->get_pending_records_ptr();
```

Comment:
There's no need in one_session_records._local_tracing_ptr here, the caller of this fn is class tracing itself, it can pass *this here via tracing& argument.

Reply:
@xemul You are not wrong but I like @bhalevy 's version more - this is much more RAII. This way we don't need to bother about the liveness of the corresponding `tracing` object in the future if, for instance we decide to make this function asynchronous for any reason.